networkx/algorithms/approximation/steinertree.py

```diff
@@ -0,0 +1,75 @@
+from itertools import combinations, chain
+
+from networkx.utils import pairwise, not_implemented_for
+import networkx as nx
+
+__all__ = ['metric_closure', 'steiner_tree']
+
+
+@not_implemented_for('directed')
+def metric_closure(G, weight='weight'):
+    """  Return the metric closure of a graph.
+
+    The metric closure of a graph G is the complete graph in which each edge is
+    weighted by the shortest path distance between the nodes in G.
+
+    Parameters
+    ----------
+    G : Networkx graph
+
+    Returns
+    -------
+    Networkx graph
+        Metric closure of the graph G.
+
+    """
+    M = nx.Graph()
+
+    for u, v in combinations(G, 2):
+        distance, path = nx.single_source_dijkstra(G, u, v, weight=weight)
+        M.add_edge(u, v, distance=distance[v], path=path[v])
+
+    return M
+
+
+@not_implemented_for('directed')
+def steiner_tree(G, terminal_nodes, weight='weight'):
+    """ Returns an approximation to minimum steiner tree of the graph G.
+
+    Parameters
+    ----------
+    G : Networkx graph
+
+    terminal_nodes : List
+                     A list of terminal nodes for which minimum steiner tree is
+                     to be found.
+
+    Returns
+    -------
+    Networkx graph
+        Approximation to the minimum steiner tree of G induced by
+        terminal_nodes.
```

Comment:
References to parameters should be surrounded by backticks, like this: ``` `terminal_nodes` ```. That way, they are rendered appropriately by the documentation builder.